includes/class-wc-deprecated-action-hooks.php

```diff
@@ -156,7 +156,7 @@ protected function trigger_hook( $old_hook, $new_callback_args ) {
 				$item     = $new_callback_args[1];
 				$order_id = $new_callback_args[2];
 				if ( is_a( $item, 'WC_Order_Item_Product' ) ) {
-					do_action( $old_hook, $item_id, $item->legacy_values, $item->legacy_cart_item_key );
+					do_action( $old_hook, $item_id, isset( $item->legacy_values ) ? $item->legacy_values : array(), isset( $item->legacy_cart_item_key ) ? $item->legacy_cart_item_key : '' );
```

Comment:
As I mentioned in the review description, I believe that the underlying problem is that here we are calling the action `woocommerce_add_order_item_meta` whenever an order is created and, before the deprecation, this action was called only when orders were created via the frontend. So I suggest that, instead of checking to see if `legacy_values` and `legacy_cart_item_key` are set to decide which values to pass to the action, we only call `do_action()` if those two properties are set. What do you think?

What I have in mind is the following (untested code):

```
if ( is_a( $item, 'WC_Order_Item_Product' ) && isset( $item->legacy_values, $item->legacy_cart_item_key ) ) {
	do_action( $old_hook, $item_id, $item->legacy_values, $item->legacy_cart_item_key );
}
```